### PR TITLE
Add arguments for parallelization to differential_expression and residualized_counts

### PR DESCRIPTION
**Pull request by jgockley62**
Fixes #152. Added `BPPARAM = BiocParallel::SnowParam(parallel::detectCores()-1)` to `variancePartition::dream` and `variancePartition::voomWithDreamWeights` function calls.

### CODE REVIEW

**Review by kelshmo — CHANGES_REQUESTED** [review]
I think this should work OK. We also need to declare `parallel` and `BiocParallel` as imports in the DESCRIPTION file. Have you tested whether this works in a fresh environment?

**Comment by jgockley62** [comment]
@kelshmo Implements pulling the number of cores from the config directly. If NULL cores is set to: ```parallel::detectCores()-1```

File: R/functions.R [hunk 1]
```diff
@@ -515,13 +515,14 @@ build_formula <- function(md, primary_variable, model_variables = NULL,
...
 #' effect. If you wish to test an interaction term, `primary_variable` can take multiple variable names.
-#'
+#' 
 #' @param cqn_counts A counts data frame normalized by CQN.
 #' @param p_value_threshold Numeric. P-values are adjusted by Benjamini and
 #' Hochberg (BH) false discovery rate (FDR). Significant genes are those with an
 #' adjusted p-value greater than this threshold.
 #' @param fold_change_threshold Numeric. Significant genes are those with a
 #' fold-change greater than this threshold.
+#' @param cores An integer of cores to specify in the parallel backend (eg. 4)
```
Comment (kelshmo) [inline R/functions.R]:
```suggestion
#' @param cores An integer of cores to specify in the parallel backend (eg. 4).
```

File: R/functions.R [hunk 2]
```diff
@@ -515,13 +515,14 @@ build_formula <- function(md, primary_variable, model_variables = NULL,
 #' Differential expression testing is performed with \code{"variancePartition::dream()"} to increase power
 #' and decrease false positives for repeated measure designs. The `primary_variable` is modeled as a fixed
 #' effect. If you wish to test an interaction term, `primary_variable` can take multiple variable names.
-#'
+#'
```
Comment (kelshmo) [inline R/functions.R]:
```suggestion
#'
```

File: vignettes/customize-config.Rmd [hunk 3]
```diff
@@ -68,6 +68,11 @@ skip model:       Optional. If TRUE, the exploratory data report is run. Model s
                   is not computed.
 force model with: Optional. Force differential expression with this user defined model
                   instead of the output of stepwise regression.
+cores:            Optional. Specify an integer of cores to use with a BiocParallel 
+                  parallel backend.  Null value results in the number of available
+                  cores minus one being used. Parallel backend ccurrently only supports
+                  BiocParallel::SnowParam(). BatchtoolsParam, MulticoreParam, 
+                  BiocParallelDoparParam, and SerialParam not currently supported.
```
Comment (kelshmo) [inline vignettes/customize-config.Rmd]:
```suggestion
                  BiocParallelDoparParam, and SerialParam are not currently supported.
```

**Review by kelshmo — CHANGES_REQUESTED** [review]
Can you run `devtools::document()` and commit the new documentation in the man/ folder? I can't test the function until you do that.

There are some small suggested changes to syntax.

**Comment by jgockley62** [comment]
@kelshmo Suggestions implemented and `devtools::document()` run with new manual pages pushed as well.

**Comment by kelshmo** [comment]
@jgockley62 `wrap_de` is missing the cores parameter. You will also need to commit the man/ files for that function.

**Comment by jgockley62** [comment]
@kelshmo whoops my bad should be added

**Review by kelshmo — APPROVED** [review]

**Comment by kelshmo** [comment]
This works! I tested omitting parameters and manually setting. Well done!

**Pull request merged**